node-client/src/web-socket-handler.ts

```diff
@@ -96,37 +104,38 @@ export class WebSocketHandler implements WebSocketInterface {
         const proto = ssl ? 'wss' : 'ws';
         const uri = `${proto}://${target}${path}`;
 
-        const opts: https.RequestOptions = {};
+        const opts: WebSocket.ClientOptions = {};
         // TODO: This doesn't set insecureSSL if skipTLSVerify is set...
         this.config.applytoHTTPSOptions(opts);
 
-        const client = new ws.client({ tlsOptions: opts });
-
         return new Promise((resolve, reject) => {
-            client.on('connect', (connection) => {
-                connection.on('message', (message: ws.IMessage) => {
-                    if (message.type === 'utf8' && message.utf8Data) {
-                        if (textHandler) {
-                            if (!textHandler(message.utf8Data)) {
-                                connection.close();
-                            }
-                        }
-                    } else if (message.type === 'binary' && message.binaryData) {
-                        if (binaryHandler) {
-                            const streamNum = message.binaryData.readInt8(0);
-                            if (!binaryHandler(streamNum, message.binaryData.slice(1))) {
-                                connection.close();
-                            }
-                        }
+            const client = new WebSocket(uri, opts);
+            let resolved = false;
+
+            client.onopen = () => {
+                resolved = true;
+                resolve(client);
+            };
+
+            client.onerror = (err) => {
+                if (!resolved) {
+                    reject(err);
+                }
+            };
+
+            client.onmessage = ({ data }: { data: WebSocket.Data }) => {
+                // TODO: support ArrayBuffer and Buffer[] data types?
+                if (typeof data === 'string') {
+                    if (textHandler && !textHandler(data)) {
+                        client.close();
+                    }
+                } else if (binaryHandler && data instanceof Buffer) {
```

Comment:
I'd suggest we keep the existing structure where the top level `if` checks only on the type of data, and the second level makes it a no-op if there is no handler for the type of data received:

```
if (data instanceof Buffer) {
  if (binaryHandler) {
    // ...
  }
}
```

This keeps the structure consistent with the string case (although of course at the moment the behaviour is the same either way).